subprojects/solver/solver-javasmt/src/main/java/hu/bme/mit/theta/solver/javasmt/JavaSMTTermTransformer.java

```diff
@@ -0,0 +1,550 @@
+                            return transformRatLit(f, (BigInteger) value);
+                        }
+                    } else if (type.isBitvectorType()) {
+                        checkArgument(value instanceof BigInteger, "Type mismatch (Expected BigInteger): " + value);
+                        return transformBvLit(f, (BigInteger) value);
+                    } else if (type.isFloatingPointType()) {
+                        checkArgument(value instanceof BigDecimal, "Type mismatch (Expected BigFloat): " + value);
+                        return transformFpLit(f, (BigDecimal) value);
+                    } else if (type.isArrayType()) {
+                        return transformArrLit(f, value, model, vars);
+                    } else if (type.isBooleanType()) {
+                        if (Boolean.TRUE.equals(value)) {
+                            return True();
+                        } else if (Boolean.FALSE.equals(value)) {
+                            return False();
+                        }
+                    }
+                    throw new JavaSMTSolverException("Not supported: " + f);
+                }
+
+                @Override
+                public Expr<?> visitFunction(Formula f, List<Formula> args, FunctionDeclaration<?> functionDeclaration) {
+                    return transformApp(f, functionDeclaration, args, model, vars);
+                }
+
+                @Override
+                public Expr<?> visitQuantifier(BooleanFormula f, Quantifier quantifier, List<Formula> boundVariables, BooleanFormula body) {
+                    return transformQuantifier(quantifier, boundVariables, model, body, vars);
+                }
+            });
+        } catch (JavaSMTSolverException e) {
+            throw new JavaSMTSolverException("Not supported: " + term, e);
+        }
+    }
+
+    ////
+
+    private Expr<?> transformIntLit(final Formula term, final BigInteger value) {
+        return Int(value);
+    }
+
+    private Expr<?> transformRatLit(final Formula term, BigInteger value) {
+        return Rat(value, BigInteger.ONE);
+    }
+
+    private Expr<?> transformRatLit(final Formula term, Rational value) {
+        return Rat(value.getNum(), value.getDen());
+    }
+
+    private Expr<?> transformArrLit(final Formula term, Object value, final Model model,
+                                    final List<Decl<?>> vars) {
+        throw new JavaSMTSolverException("Not supported: " + term);
+    }
+
+    private Expr<?> transformBvLit(final Formula term, BigInteger value) {
+        final var bvNum = (BitvectorFormula) term;
+        BitvectorType formulaType = (BitvectorType) context.getFormulaManager().getFormulaType(bvNum);
+
+        return BvUtils.bigIntegerToNeutralBvLitExpr(value, formulaType.getSize());
+    }
+
+    private Expr<?> transformFpLit(final Formula term, BigDecimal value) {
+        final var fpNum = (FloatingPointFormula) term;
```

Comment:
If https://github.com/sosy-lab/java-smt/issues/360 fixes this, we should implement it correctly.